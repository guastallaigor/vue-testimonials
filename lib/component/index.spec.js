/*eslint no-undef: "off"*/
import { mount } from '@vue/test-utils'
import Component from './VueTestimonials.vue'

describe('When I create the VueTestimonials component', () => {
  const items = [
    {
      title: 'Excelent',
      content:
        '"I just updated my site and it was so simple! Thanks for being so awesome!"',
      image: 'https://i.pravatar.cc/160?img=11',
      author: 'Jason',
    },
    {
      title: 'Outstanding',
      content:
        '"She was thoughtful during the entire process and exceed my expectations!"',
      image: 'https://i.pravatar.cc/160?img=31',
      author: 'Emily',
    },
    {
      title: 'The best',
      content:
        '"My new site is so much faster and easier to work with. Thanks guys!"',
      image: 'https://i.pravatar.cc/160?img=45',
      author: 'Amelia',
    },
  ]

  const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms))

  const createComponent = (props = {}, mocks = {}) => {
    return mount(Component, {
      props,
      global: {
        mocks,
      },
    })
  }

  it('should test clicking on the small ball', async () => {
    const wrapper = createComponent({ items })
    expect(wrapper.vm.active).toBe(1)
    wrapper.vm.goToTestimonial(1)
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.active).toBe(2)
  })

  it('should go to the next testimonial', async () => {
    const wrapper = createComponent({ items })
    expect(wrapper.vm.active).toBe(1)
    wrapper.vm.nextTestimonial()
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.active).toBe(2)
  })

  it('should go to the previous testimonial', async () => {
    const wrapper = createComponent({ items })
    expect(wrapper.vm.active).toBe(1)
    wrapper.vm.previousTestimonial()
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.active).toBe(3)
  })

  it('should stop the carousel', async () => {
    const wrapper = createComponent({ items, interval: 1 })
    expect(wrapper.vm.active).toBe(1)
    wrapper.vm.clearCarousel()
    await timeout(100)
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.active).toBe(1)
  })

  it('should match default component snapshot', () => {
    const wrapper = createComponent({ items })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
