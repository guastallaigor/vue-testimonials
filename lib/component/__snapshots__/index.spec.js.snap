// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I create the VueTestimonials component should match default component snapshot 1`] = `
"<section class=\\"vue-testimonials\\">
  <div class=\\"flex-row flex flex-nowrap text-white mt-6 pb-10 lg:pb-20 mb-40\\">
    <div class=\\"mx-auto\\">
      <div>
        <div class=\\"relative\\">
          <div class=\\"w-full\\">
            <div class=\\"item transition-opacity duration-300 ease-in-out px-16 lg:px-0 active\\">
              <div class=\\"flex justify-center flex-wrap items-center text-center\\"><img src=\\"https://i.pravatar.cc/160?img=11\\" class=\\"mr-3 lazyload rounded-full object-cover image w-40 h-40\\" alt=\\"Jason\\">
                <div class=\\"media-body w-full\\">
                  <h1 class=\\"mt-12 text-2xl uppercase tracking-wider font-normal\\">Excelent</h1>
                  <h2 class=\\"mt-5 text-lg\\">\\"I just updated my site and it was so simple! Thanks for being so awesome!\\"</h2>
                  <div class=\\"mt-5 text-md font-bold select-none\\">Jason</div>
                </div>
              </div>
            </div>
            <div class=\\"item transition-opacity duration-300 ease-in-out px-16 lg:px-0 inactive\\">
              <div class=\\"flex justify-center flex-wrap items-center text-center\\"><img src=\\"https://i.pravatar.cc/160?img=31\\" class=\\"mr-3 lazyload rounded-full object-cover image w-40 h-40\\" alt=\\"Emily\\">
                <div class=\\"media-body w-full\\">
                  <h1 class=\\"mt-12 text-2xl uppercase tracking-wider font-normal\\">Outstanding</h1>
                  <h2 class=\\"mt-5 text-lg\\">\\"She was thoughtful during the entire process and exceed my expectations!\\"</h2>
                  <div class=\\"mt-5 text-md font-bold select-none\\">Emily</div>
                </div>
              </div>
            </div>
            <div class=\\"item transition-opacity duration-300 ease-in-out px-16 lg:px-0 inactive\\">
              <div class=\\"flex justify-center flex-wrap items-center text-center\\"><img src=\\"https://i.pravatar.cc/160?img=45\\" class=\\"mr-3 lazyload rounded-full object-cover image w-40 h-40\\" alt=\\"Amelia\\">
                <div class=\\"media-body w-full\\">
                  <h1 class=\\"mt-12 text-2xl uppercase tracking-wider font-normal\\">The best</h1>
                  <h2 class=\\"mt-5 text-lg\\">\\"My new site is so much faster and easier to work with. Thanks guys!\\"</h2>
                  <div class=\\"mt-5 text-md font-bold select-none\\">Amelia</div>
                </div>
              </div>
            </div>
          </div>
          <div class=\\"mt-5 w-full fill-current flex justify-center items-center flex-row z-20 relative\\">
            <ball.svg class=\\"overflow-hidden cursor-pointer hover:opacity-70 transition-opacity ease-in-out duration-300 mr-1 ball text-gray-600\\" style=\\"width: 20px; height: 20px;\\"></ball.svg>
            <ball.svg class=\\"overflow-hidden cursor-pointer hover:opacity-70 transition-opacity ease-in-out duration-300 mr-1 ball empty-circle\\" style=\\"width: 20px; height: 20px;\\"></ball.svg>
            <ball.svg class=\\"overflow-hidden cursor-pointer hover:opacity-70 transition-opacity ease-in-out duration-300 mr-1 ball empty-circle\\" style=\\"width: 20px; height: 20px;\\"></ball.svg>
          </div><a class=\\"absolute top-1/2 left-0\\" role=\\"button\\" rel=\\"noopener noreferrer\\" aria-label=\\"Previous arrow button\\" href=\\"#\\">
            <icon-chevronleft-lg.svg style=\\"color: rgb(255, 255, 255); width: 64px; height: 64px;\\" class=\\"hover:opacity-70 transition-opacity ease-in-out duration-500 lg:w-full arrow\\"></icon-chevronleft-lg.svg>
          </a><a class=\\"absolute top-1/2 right-0\\" role=\\"button\\" href=\\"#\\" aria-label=\\"Next arrow button\\" rel=\\"noopener noreferrer\\">
            <icon-chevronright-lg.svg style=\\"color: rgb(255, 255, 255); width: 64px; height: 64px;\\" class=\\"hover:opacity-70 transition-opacity ease-in-out duration-500 lg:w-full arrow\\"></icon-chevronright-lg.svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</section>"
`;
